src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -702,52 +759,14 @@ private void LibraryLoaded(object sender, LibraryServices.LibraryLoadedEventArgs
             // Load all functions defined in that library.
             AddZeroTouchNodesToSearch(LibraryServices.GetFunctionGroups(newLibrary));
         }
-
-        /// <summary>
-        /// This event handler is invoked when DynamoScheduler changes the state 
-        /// of an AsyncTask object. See TaskStateChangedEventArgs.State for more 
-        /// details of these state changes.
-        /// </summary>
-        /// <param name="sender">The scheduler which raised the event.</param>
-        /// <param name="e">Task state changed event argument.</param>
-        /// 
-        private void OnAsyncTaskStateChanged(DynamoScheduler sender, TaskStateChangedEventArgs e)
-        {
-            switch (e.CurrentState)
-            {
-                case TaskStateChangedEventArgs.State.ExecutionStarting:
-                    if (e.Task is UpdateGraphAsyncTask)
-                        ExecutionEvents.OnGraphPreExecution();
-                    break;
-
-                case TaskStateChangedEventArgs.State.ExecutionCompleted:
-                    if (e.Task is UpdateGraphAsyncTask)
-                    {
-                        // Record execution time for update graph task.
-                        long start = e.Task.ExecutionStartTime.TickCount;
-                        long end = e.Task.ExecutionEndTime.TickCount;
-                        var executionTimeSpan = new TimeSpan(end - start);
-
-                        InstrumentationLogger.LogAnonymousTimedEvent(
-                            "Perf",
-                            e.Task.GetType().Name,
-                            executionTimeSpan);
-
-                        Debug.WriteLine(String.Format(Resources.EvaluationCompleted, executionTimeSpan));
-
-                        ExecutionEvents.OnGraphPostExecution();
-                    }
-                    break;
-            }
-        }
-
```

Comment:
All of these methods are moved into `OnEvaluationStarted` and `OnEvaluationCompleted`.